core/src/main/java/hudson/Launcher.java

```diff
@@ -142,6 +143,20 @@ public Computer getComputer() {
     }
 
     /**
+     * Gets the IP address of the host running the jenkins slave agent which do control this Launcher
+     * <p>
+     * In most cases, slave agent do run on localhost, but within a containerized architecture, the
+     * slave agent hostname may be a distinct one. A {@link LauncherDecorator} responsible to setup
+     * such an architecture could then decorate this method to return the adequate slave agent host address.
```

Comment:
Yet existing `Launcher.decorateBy*` methods have not been updated to delegate this method. They should be made to use `DecoratedLauncher`, which must then override this.